src/test/java/wolox/training/BookRepositoryIntegrationTest.java

```diff
@@ -0,0 +1,88 @@
+package wolox.training;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+public class BookRepositoryIntegrationTest {
+
+
+    @Autowired
+    private Book book;
+    @Autowired
+    private Book bookSaved;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    @Before
+    public void setUp(){
+        book = new Book();
+        book.setGenre("Genre");
+        book.setAuthor("Test");
+        book.setImage("image.png");
+        book.setTitle("Book 1");
+        book.setSubtitle("Sub_1");
+        book.setPublisher("Publisher-1");
+        book.setPages(130);
+        book.setYear("2018");
+        book.setIsbn("987654321");
+        bookSaved = new Book();
+        bookSaved.setGenre("Genre 2");
+        bookSaved.setAuthor("Test 2");
+        bookSaved.setImage("image.png");
+        bookSaved.setTitle("Book 2");
+        bookSaved.setSubtitle("Sub_2");
+        bookSaved.setPublisher("Publisher-2");
+        bookSaved.setPages(120);
+        bookSaved.setYear("2012");
+        bookSaved.setIsbn("9876322321");
+        bookSaved = bookRepository.save(bookSaved);
```

Comment:
Use entityManager instead. (  @Autowired private TestEntityManager entityManager;)
```
entityManager.persist(book);
entityManager.flush();
```

Reply:
Change the method to save entity instead repository